### PR TITLE
New emails for sharebymail

### PR DESCRIPTION
**Pull request by MorrisJobke**
* I added a new method to allow adding a single button with no width constraints
* escape the texts of the heading, body and button text with `htmlspecialchars` (is this enough, @LukasReschke ) - this was needed to show also file names in there and not cause code injection (the escaping is only done in the HTML emails and not in the text emails
* migrate share email and the password notification email

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Followup to #4304 and #4244

**Comment by MorrisJobke** [comment]
Before:

bildschirmfoto 2017-04-11 um 16 25 44

After:

bildschirmfoto 2017-04-11 um 16 26 35
bildschirmfoto 2017-04-11 um 16 25 49

**Comment by MorrisJobke** [comment]
@nextcloud/designers I'm also open for new wordings of the emails, because it's a lot of duplicate text. Maybe you have better ideas. cc @Espina2

**Review by rullzer — APPROVED** [review]
Awesome stuff. Lets do this!

**Comment by LukasReschke** [comment]
Rebased upon master.

**Review by schiessle — CHANGES_REQUESTED** [review]
The HTML mail looks awesome! But the text mail looks strange:

````
admin shared »welcome.txt« with you

admin shared »welcome.txt« with you. Click the button below to open it.

Open »welcome.txt«: http://localhost/master/index.php/s/RPGlwrxTJr2lrm5
--
Nextcloud - a safe home for all your data
This is an automatically generated email, please do not reply.
````

Can we fix this?

**Comment by nickvergessen** [comment]
i will fix

**Comment by LukasReschke** [comment]
> escape the texts of the heading, body and button text with htmlspecialchars (is this enough, @LukasReschke ) - this was needed to show also file names in there and not cause code injection (the escaping is only done in the HTML emails and not in the text emails

Not entirely sure about that. For example in the activity app we would have an email that uses something like:

> $emailTemplate->addBodyText($l->t('There was some activity at %s', [$this->urlGenerator->getAbsoluteURL('/')]));

In this case having HTML allowed here for URLs (and a separate text for plain text) may be very useful. Also I think sometimes some stuff like `<strong>` may be useful as well. Thoughts?

**Comment by LukasReschke** [comment]
Also HTML may be useful when embedding icons like the activity state icons. Just a thought though :-)

**Comment by schiessle** [comment]
I fixed the footer for the plain text mails.

The plain text mail looks now better, also with the changes by @nickvergessen :

mail

Although I think the "Click the button below to open it." still sounds a bit strange for a plain-text mail. I think the best would be to create two separate templates. This way we can make sure that we have a useful text in both cases and that we have reasonable line breaks (something <= 80 chars).

**Comment by nickvergessen** [comment]
@schiessle `addBodyText` has two parameters, just make use of it.

**Comment by Espina2** [comment]
I think you can more information like that. @MorrisJobke 

shared folder

**Comment by enoch85** [comment]
@Espina2 Just plain perfection.

**Comment by schiessle** [comment]
@nickvergessen I saw it. The "problem" is that we then have to split it up in two "addBodyText()" calls which will result in a line break, right?

**Comment by schiessle** [comment]
As expected it looks rather ugly on the html mail if it put in in two "addBodyText()" calls because of the line breaks:

mail

**Comment by nickvergessen** [comment]
Not sure what you did, but I made two little commits and it looks good

**Review by schiessle — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Tested and works nice 👍

**Comment by MorrisJobke** [comment]
@Espina2 Looks nice, but something for a followup. First of all we should get rid of the super ugly way. ;)

File: lib/private/Mail/EMailTemplate.php [hunk 1]
```diff
@@ -461,7 +461,7 @@ public function addFooter($text = '') {
 
 		$this->htmlBody .= vsprintf($this->footer, [$text]);
 		$this->htmlBody .= $this->tail;
-		$this->plainBody .= '--' . PHP_EOL;
+		$this->plainBody .= PHP_EOL . '-- ' . PHP_EOL;
```
Comment (MorrisJobke) [inline lib/private/Mail/EMailTemplate.php]: When you update stuff like this, then you should also update the unit tests. Let me do this.

**Comment by MorrisJobke** [comment]
I fixed the unit tests. Looks good now from my side 👍

**Pull request merged**